File: app/src/main/java/tourguide/instrumentation/test/ToolTipMeasureTestActivity.java
package tourguide.instrumentation.test;

import android.app.Activity;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.app.ActionBarActivity;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;

import tourguide.tourguide.Overlay;
import tourguide.tourguide.Pointer;
import tourguide.tourguide.ToolTip;
import tourguide.tourguide.TourGuide;
import tourguide.tourguidedemo.R;


public class ToolTipMeasureTestActivity extends ActionBarActivity {
    public TourGuide mTutorialHandler;
    public Activity mActivity;
    public static final String TOOLTIP_NUM = "tooltip_num";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        /* Get parameters from main activity */
        Intent intent = getIntent();
        int tooltip_num = intent.getIntExtra(TOOLTIP_NUM, 1);

        super.onCreate(savedInstanceState);
        mActivity = this;
        int gravity;

        if (tooltip_num == 1) {
            setContentView(R.layout.activity_tooltip_gravity_i);
            gravity = Gravity.RIGHT | Gravity.BOTTOM;
        } else if (tooltip_num == 2) {
            setContentView(R.layout.activity_tooltip_gravity_ii);
            gravity = Gravity.LEFT | Gravity.BOTTOM;
        } else if (tooltip_num == 3) {
            setContentView(R.layout.activity_tooltip_gravity_iii);
            gravity = Gravity.LEFT | Gravity.TOP;
        } else {
            setContentView(R.layout.activity_tooltip_gravity_iv);
            gravity = Gravity.RIGHT | Gravity.TOP;
        }
        Button button = (Button)findViewById(R.id.button);

        ToolTip toolTip = new ToolTip().
                setTitle("Welcome!").
                setDescription("This is a really really long title....This is a really really long title....This is a really really long title....This is a really really long title....This is a really really long title....This is a really really long title....This is a really really long title....").
                setBackgroundColor(Color.parseColor("#2980b9")).
                setTextColor(Color.parseColor("#FFFFFF")).
                setGravity(gravity).
                setShadow(true);

        mTutorialHandler = TourGuide.init(this).with(TourGuide.Technique.Click)
                .setPointer(new Pointer())
                .setToolTip(toolTip)
                .setOverlay(new Overlay())
                .playOn(button);

        button.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                mTutorialHandler.cleanUp();
            }
        });
    }
}

File: app/src/main/java/tourguide/instrumentation/test/ToolTipMeasureTestActivity.kt
package tourguide.instrumentation.test

import android.graphics.Color
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Gravity
import kotlinx.android.synthetic.main.activity_tooltip_gravity_i.*
import tourguide.tourguide.Overlay
import tourguide.tourguide.Pointer
import tourguide.tourguide.TourGuide
import tourguide.tourguidedemo.R


class ToolTipMeasureTestActivity : AppCompatActivity() {
    lateinit var tourGuide: TourGuide

    override fun onCreate(savedInstanceState: Bundle?) {
        /* Get parameters from main activity */
        val intent = intent
        val tooltipNumber = intent.getIntExtra(TOOLTIP_NUM, 1)

        super.onCreate(savedInstanceState)
        val gravity = when (tooltipNumber) {
            1 -> {
                setContentView(R.layout.activity_tooltip_gravity_i)
                Gravity.END or Gravity.BOTTOM
            }
            2 -> {
                setContentView(R.layout.activity_tooltip_gravity_ii)
                Gravity.START or Gravity.BOTTOM
            }
            3 -> {
                setContentView(R.layout.activity_tooltip_gravity_iii)
                Gravity.START or Gravity.TOP
            }
            else -> {
                setContentView(R.layout.activity_tooltip_gravity_iv)
                Gravity.END or Gravity.TOP
            }
        }

        tourGuide = TourGuide.create(this) {
            overlay { Overlay() }
        }.playOn(button) {
            pointer { }
            toolTip {
                title { "Welcome!" }
                description { "This is a really really long title....This is a really really long title....This is a really really long title....This is a really really long title....This is a really really long title....This is a really really long title....This is a really really long title...." }
                backgroundColor { Color.parseColor("#2980b9") }
                textColor { Color.parseColor("#FFFFFF") }
                gravity { gravity }
                shadow { true }
            }
        }.show()

        button.setOnClickListener { tourGuide.cleanUp() }
    }

    companion object {
        const val TOOLTIP_NUM = "tooltip_num"
    }
}
